cmake/core-files.cmake

```diff
@@ -531,6 +531,7 @@ set(MBGL_CORE_FILES
     include/mbgl/util/immutable.hpp
     include/mbgl/util/indexed_tuple.hpp
     include/mbgl/util/interpolate.hpp
+    include/mbgl/util/io.hpp
```

Comment:
Sure - my rationale when doing this was because our `read_file` implementation relies solely on C++ standard library.